Simplenote/src/main/java/com/automattic/simplenote/TagDialogFragment.java

```diff
@@ -87,12 +79,62 @@ public void onClick(DialogInterface dialog, int whichButton) {
         builder.setView(view);
         mDialogEditTag = builder.create();
         mDialogEditTag.setOnShowListener(this);
+
+        mClickListenerNegativeRename = new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                dismiss();
+            }
+        };
+
+        mClickListenerPositiveRename = new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String tagNew = getTagNew();
+
+                if (tagNew.equals(mTagOld)) {
+                    dismiss();
+                }
+
+                int index = mTag.hasIndex() ? mTag.getIndex() : mBucketTag.count();
+                boolean isRenamingToLexicalTag = TagUtils.hashTag(tagNew).equals(TagUtils.hashTag(mTagOld));
+                boolean hasCanonicalTag = TagUtils.hasCanonicalOfLexical(mBucketTag, tagNew);
+
+                if (hasCanonicalTag && !isRenamingToLexicalTag) {
+                    String tagCanonical = TagUtils.getCanonicalFromLexical(mBucketTag, tagNew);
+                    showDialogErrorConflict(tagCanonical, mTagOld);
+                    return;
+                }
+
+                tryToRenameTag(tagNew, index);
+            }
+        };
+
+        mClickListenerNegativeConflict = new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                showDialogRenameTag();
```

Comment:
Question (❓ ): This is a new behavior, isn't it?

Reply:
Yes, tapping the negative button in the dialog, which is currently ***Cancel***, reverts to the ***Rename Tag*** view.  Since the dialog was updated to show/hide views between ***Rename Tag*** and ***Tag Conflict*** rather than using separate dialogs, I chose to add that behavior.  @SylvesterWilmott, what do you think of that interaction (i.e. returning to the previous dialog view allowing the user to rename the tag again rather than dismissing the dialog completely)?  If we keep that interaction, should we change the ***Cancel*** button to ***Back*** or something similar so be a little clearer to the user what will happen?  See the animation below for illustration.

<kbd><a href="https://user-images.githubusercontent.com/3827611/101591324-89421e00-39a9-11eb-80ce-1da9ebdee29d.gif"><img src="https://user-images.githubusercontent.com/3827611/101591324-89421e00-39a9-11eb-80ce-1da9ebdee29d.gif" width="300"></a></kbd>

Reply:
The interaction of returning to the renaming dialog from the confirmation dialog is ideal. Ideally the button would describe the action such as "Choose another name" but this label is far too long. I think in this case "Back" works just fine.